### ISSUE TITLE
Can't find completions in wandb and using IS correction causes loss=0.0

### ISSUE THREAD

**Issue by Aurora-1412** [huggingface/trl#4767]
### Reproduction

```bash
accelerate launch \
    --config_file ./configs/deepspeed_zero3.yaml \
    /home/xxx/code/train_guess_words/train/test_train.py \
    --model_name_or_path /data/xxx/models/Qwen3-4B \
    --output_dir GRPO-Qwen3-4B \
    --learning_rate 3e-6 \
    --max_steps 300 \
    --gradient_checkpointing \
    --dtype bfloat16 \
    --max_completion_length 1024 \
    --use_vllm \
    --vllm_mode server \
    --log_completions \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --num_generations 7 \
    --warmup_steps 5 \
    --temperature 0.8 \
    --top_p 0.9 \
    --beta 0.02 \
    --epsilon 0.2 \
    --epsilon_high 0.3 \
    --scale_rewards group \
    --loss_type dapo \
    --vllm_mode server \
    --vllm_server_base_url http://localhost:8108 \
    --vllm_server_port 8108 \
    --vllm_tensor_parallel_size 1 \
    --vllm_enable_sleep_mode \
    --vllm_importance_sampling_correction False \
    --log_completions \
    --num_completions_to_print 3 \
    --log_level info \
    --logging_steps 1 \
    --report_to wandb \
    --save_strategy steps \
    --save_steps 50 \
    --eval_strategy steps \
    --eval_steps 50
```

```python
if __name__ == "__main__":
    parser = TrlParser((ScriptArguments, GRPOConfig, ModelConfig))
    script_args, training_args, model_args = parser.parse_args_and_config()

    ################
    # Model
    ################
    dtype = model_args.dtype if model_args.dtype in ["auto", None] else getattr(torch, model_args.dtype)
    training_args.model_init_kwargs = dict(
        revision=model_args.model_revision,
        attn_implementation=model_args.attn_implementation,
        dtype=dtype,
    )
    quantization_config = get_quantization_config(model_args)
    if quantization_config is not None:
        # Passing None would not be treated the same as omitting the argument, so we include it only when valid.
        training_args.model_init_kwargs["device_map"] = get_kbit_device_map()
        training_args.model_init_kwargs["quantization_config"] = quantization_config

    ################
    # GRPO config
    ################
    training_args.chat_template_kwargs = {"enable_thinking": True}
    

    ################
    # Dataset
    ################
    from datasets import load_from_disk
    dataset = load_from_disk("/home/xxx/code/train_guess_words/guess_word_dataset_conversation")

    train_dataset = dataset["train"]
    eval_dataset = dataset["test"] if training_args.eval_strategy != "no" else None

    ################
    # Training
    ################
    training_args.reward_weights = [0.05, 0.05, 0.2, 1]
    trainer = GRPOTrainer(
        model='/data/liyihang/models/Qwen3-4B',
        args=training_args,
        reward_funcs=[think_format_reward, answer_format_reward, length_reward, accuracy_reward],
        train_dataset=train_dataset,
        eval_dataset=eval_dataset,
    )

    trainer.train()
```
wandb outputs: I can see responses in logs, but there's no completion.

<img width="1352" height="282" alt="Image" src="https://github.com/user-attachments/assets/04377d20-7cce-4081-b486-b814264a06a5" />

By the way, 
I have to set `--vllm_importance_sampling_correction False`

If it's true, `train/loss` is always zero. `importance_sampling_ratio` is also 0, (`train/frac_reward_zero_std` not always 0)

### System Info

macOS python3.10
torch 2.8.0
transformers Version: 4.57.1
trl Version: 0.26.2
wandb Version: 0.23.1

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by JeffLee1874** [comment]
same issue, when vllm_importance_sampling_correction is false, all is normal

**Comment by Aurora-1412** [comment]
I have tested fp16 and bf16，using fp16 may can solve this problem.&nbsp;



---Original---
From: ***@***.***&gt;
Date: Wed, Apr 8, 2026 00:10 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [huggingface/trl] Can't find completions in wandb and using IS correction causes loss=0.0 (Issue #4767)


JeffLee1874 left a comment (huggingface/trl#4767)
 
same issue, when vllm_importance_sampling_correction is false, all is normal

**Comment by JeffLee1874** [comment]
> I have tested fp16 and bf16，using fp16 may can solve this problem.&nbsp;
> […](#)
> 
> ---Original---
> From: ***@***.***&gt;
> Date: Wed, Apr 8, 2026 00:10 AM
> To: ***@***.***&gt;;
> Cc: ***@***.******@***.***&gt;;
> Subject: Re: [huggingface/trl] Can't find completions in wandb and using IS correction causes loss=0.0 (Issue [#4767](https://github.com/huggingface/trl/issues/4767))
> 
> 
> JeffLee1874 left a comment ([huggingface/trl#4767](https://github.com/huggingface/trl/issues/4767))
> 
> same issue, when vllm_importance_sampling_correction is false, all is normal
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

Have you try vllm_importance_sampling_mode=token_mask?

**Comment by albertvillanova** [comment]
This is essentially the same question as your other thread #4772.

Closing as a duplicate of #4772.
